src/plugins/undoRedo/__tests__/hooks.spec.js

```diff
@@ -0,0 +1,182 @@
+describe('UndoRedo', () => {
+  const id = 'testContainer';
+
+  beforeEach(function() {
+    this.$container = $(`<div id="${id}"></div>`).appendTo('body');
+  });
+
+  afterEach(function() {
+    if (this.$container) {
+      destroy();
+      this.$container.remove();
+    }
+  });
+
+  describe('Hooks', () => {
+    it('should fire a `beforeUndo` hook after the undo process begins', (done) => {
```

Comment:
I encourage you to use `async`/`await` grammar instead of trick with the `done`/`setTimeout` approach.

Reply:
It's not my code, but I can refactor it.

Reply:
It'd be great 💪

Reply:
Changed within https://github.com/handsontable/handsontable/pull/7827/commits/7ed41e2f2560e46f18993533732b6d5629885b96.